src/Deserialization/OperandFactory.cs

```diff
@@ -0,0 +1,143 @@
+                LLVMValueKind.LLVMConstantIntValueKind =>
+                    CreateConstantInteger(operand),
+                LLVMValueKind.LLVMConstantFPValueKind =>
+                    CreateConstantFloat(operand),
+                LLVMValueKind.LLVMConstantPointerNullValueKind =>
+                    new ConstantNull(),
+                LLVMValueKind.LLVMConstantTokenNoneValueKind =>
+                    new Unsupported("none"),
+                LLVMValueKind.LLVMMetadataAsValueValueKind =>
+                    new Metadata(),
+                LLVMValueKind.LLVMInlineAsmValueKind =>
+                    new Unsupported("asm"),
+                LLVMValueKind.LLVMInstructionValueKind =>
+                    new Variable((InstructionId) _idFactory.GetOrCreate(operand.Handle)),
+                LLVMValueKind.LLVMPoisonValueValueKind =>
+                    new Unsupported("poison"),
+                _ =>
+                    throw new Exception("Operand type is unknown.")
+            };
+        }
+
+        private IOperand CreateConstantStruct(LLVMValueRef operand, IInstructionFactory instructionFactory)
+        {
+            return new ConstantStruct(
+                operand.TypeOf.GetStoreSize(_targetData).ToBits(),
+                operand.GetOperands()
+                    .Select((o, i) => new StructElement(
+                        operand.TypeOf.GetElementOffset(_targetData, (uint) i).ToBits(),
+                        Create(o, instructionFactory)))
+                    .ToArray());
+        }
+
+        private IOperand CreateConstantAggregate(LLVMValueRef operand, IInstructionFactory instructionFactory)
+        {
+            return new ConstantSequence(
+                operand.TypeOf.GetStoreSize(_targetData).ToBits(),
+                operand.GetOperands()
+                    .Select(o => Create(o, instructionFactory))
+                    .ToArray());
+        }
+
+        private IOperand CreateConstantData(LLVMValueRef operand, IInstructionFactory instructionFactory)
+        {
+            return new ConstantSequence(
+                operand.TypeOf.GetStoreSize(_targetData).ToBits(),
+                operand.GetConstants()
+                    .Select(o => Create(o, instructionFactory))
+                    .ToArray());
+        }
+
+        private IOperand CreateConstantInteger(LLVMValueRef operand)
+        {
+            var size = operand.TypeOf.GetSize(_targetData);
+            var value = operand.GetValue();
+
+            return new ConstantInteger(
+                size,
+                size == Bits.One && bool.TryParse(value, out var boolean)
+                    ? boolean
+                        ? BigInteger.One
+                        : BigInteger.Zero
+                    : BigInteger.Parse(value));
+        }
+
+        private IOperand CreateConstantFloat(LLVMValueRef operand)
+        {
+            var size = operand.TypeOf.GetSize(_targetData);
+            var value = operand.GetValue();
```

Comment:
This is the final use of the string nonsense ... and you know where this goes and all the chaos that results.